components/greaselion/browser/greaselion_service.h

```diff
@@ -0,0 +1,35 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_GREASELION_BROWSER_GREASELION_SERVICE_H_
+#define BRAVE_COMPONENTS_GREASELION_BROWSER_GREASELION_SERVICE_H_
+
+#include <string>
+#include <vector>
+
+#include "base/macros.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "url/gurl.h"
+
+namespace greaselion {
+
+enum GreaselionFeature { REWARDS = 0, TWITTER_TIPS, LAST_FEATURE };
```

Comment:
I don't think we want an enum here because that still forces the greaselion component to know about every possible feature. I think better to just key it on a string value

Reply:
we could keep `const char[]` for the possible features, but I think we should keep that in brave/browser and not here because Greaselion shouldn't have to know about the list of possible features

Reply:
Following our conversation on Slack, I still think an `enum` makes the most sense here, and that going more generic is overkill. Greaselion feature flags are not open-ended; they are a defined, documented, finite list of preconditions to control rule matching. Since the end result of a rule matching is that executable script is injected into a page, it is especially important that all the steps governing the process be as explicit and understandable as possible. We do not have the ability to add new feature flags without a browser code update, because there will have to be matching code to toggle the flag in the controlling service. No one will be, or ought to be, extending this service other than Brave developers who could add a flag to the `enum` list. Supporting theoretical downstream developers who wish to extend this code is a non-goal.

At your request, I removed all code from `RewardsServiceImpl` in lieu of an observer, and I changed the `GreaselionServiceImpl` constructor to loop through all feature flags instead of initializing each of them individually, to future-proof against developers adding new feature flags but forgetting to initialize them explicitly.